### PR TITLE
Make `_stable_1d_sort(nb)` optional

### LINKED ISSUE #195: AUC computation is incorrect for tensors with >2048 elements.
## 🐛 Bug

When `auc` is called with `reorder=True`, a stable sort is performed on the input tensors using `_stable_1d_sort` ([code](https://github.com/PyTorchLightning/metrics/blob/master/torchmetrics/functional/classification/auc.py#L38)). However, `_stable_1d_sort` returns at most `2048` elements if `nb` is not provided ([code](https://github.com/PyTorchLightning/metrics/blob/master/torchmetrics/utilities/data.py#L179)). This leads to `auc` computation being incorrect.

### PR DESCRIPTION
**Pull request by carmocca**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [n/a] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  

## What does this PR do?
Fixes #195

### CODE REVIEW

File: torchmetrics/utilities/data.py [hunk 1]
```diff
@@ -172,12 +172,13 @@ def _stable_1d_sort(x: torch, nb: int = 2049):
     if x.ndim > 1:
         raise ValueError('Stable sort only works on 1d tensors')
     n = x.numel()
-    if n < nb:
-        x_max = x.max()
-        x = torch.cat([x, (x_max + 1) * torch.ones(nb - n, dtype=x.dtype, device=x.device)], 0)
+    if nb is not None:
```
Comment (EugenHotaj) [inline torchmetrics/utilities/data.py:175]: IMO we should just remove `nb` option. I don't see when you'd ever want it to be anything other than `2049`. The padding is just a hack to get torch to sort the array in a stable manner (look at the docstring).

File: tests/classification/test_auc.py [hunk 2]
```diff
@@ -76,3 +78,12 @@ def test_auc_functional(self, x, y):
 def test_auc(x, y, expected):
     # Test Area Under Curve (AUC) computation
     assert auc(tensor(x), tensor(y), reorder=True) == expected
+
+
+@pytest.mark.parametrize("nb", (None, 5, 15))
+def test_stable_1d_sort(nb):
```
Comment (maximsch2) [inline tests/classification/test_auc.py:84]: in your test, you are actually not testing stability at all, you are just testing if the sort is working at all. Can you try seeing if sorting an array of 8 zeros (and other equal values) produces indices=[0,1,2,3,..7] (so no changes are made by sort, meaning it's stable)

**Comment by Borda** [comment]
@carmocca I think this won't be needed after merging https://github.com/PyTorchLightning/metrics/pull/197#discussion_r618851691
